pkg/services/sqlstore/quota.go

```diff
@@ -231,7 +235,9 @@ func UpdateUserQuota(cmd *models.UpdateUserQuotaCmd) error {
 
 func GetGlobalQuotaByTarget(query *models.GetGlobalQuotaByTargetQuery) error {
 	// get quota used.
-	rawSQL := fmt.Sprintf("SELECT COUNT(*) as count from %s", dialect.Quote(query.Target))
+	rawSQL := fmt.Sprintf("SELECT COUNT(*) AS count FROM %s %s",
+		dialect.Quote(query.Target),
+		query.TargetConditions)
```

Comment:
ah this would not work here because the global quota does not use any `WHERE`s beforehand (unlike above with `WHERE user_id=?` or `WHERE org_id=?`) so it'd be `SELECT COUNT(*) AS count FROM dashboard AND is_folder = 0`.

Hmm. I suppose the cleanest way to do it would be to just set the conditions (i.e. `is_folder = 0`)  for the QuotaScope (and eventually the `Get*QuotaByTargetQuery`), and then in each place insert this condition with the appropriate operators.

So here it would be something like:

```
if query.TargetConditons != "" {
  targetConditions := fmt.Sprintf("WHERE %s", query.TargetConditions)
} else {
  targetConditions := ""
}

rawSQL := fmt.Sprintf("SELECT COUNT(*) AS count FROM %s %s", dialect.Quote(query.Target), targetConditions)
```

Reply:
Whereas above (e.g. for `GetOrgQuotaByTarget`) it would be:

```
if query.TargetConditons != "" {
  targetConditions := fmt.Sprintf("AND %s", query.TargetConditions)
} else {
  targetConditions := ""
}

rawSQL := fmt.Sprintf("SELECT COUNT(*) AS count FROM %s WHERE org_id=? %s", dialect.Quote(query.Target), targetConditions)
```

Reply:
I don't love it, but I don't hate it. Let me know if you see a better approach.

Reply:
Oh actually, I could pass the models.ReqContext at 

```
func (qs *QuotaService) getQuotaScopeConditions(target string) string {
```

and return the appropriate global vs non-global string there.

Reply:
Not sure. One way would be to add `WHERE 1=1` :smile:

Reply:
Haha I did consider that but I figured `pr/external` wouldn't get that by 😉.

Reply:
Though honestly, it could be cleaner to have that in the query than having the `WHERE` quite a way downstream. Unless I can think of a nicer approach, I will take this approach - you or another reviewer can then chime in if you think of a nicer approach.